open_mastr/utils/helpers.py

```diff
@@ -410,3 +429,329 @@ def data_to_include_tables(data: list, mapping: str = None) -> list:
+                            renaming[addit_data_type]["columns"],
+                            renaming[addit_data_type]["suffix"],
+                        )
+                    )
+
+            query_tech = Query(subtables, session=session)
+
+            # Define joins based on available tables for data and user input
+            if "unit_data" in orm_tables:
+                query_tech = query_tech.join(
+                    orm_tables["unit_data"],
+                    orm.BasicUnit.EinheitMastrNummer == orm_tables["unit_data"].EinheitMastrNummer,
+                    isouter=True,
+                )
+            if "eeg_data" in orm_tables:
+                query_tech = query_tech.join(
+                    orm_tables["eeg_data"],
+                    orm.BasicUnit.EegMastrNummer == orm_tables["eeg_data"].EegMastrNummer,
+                    isouter=True,
+                )
+            if "kwk_data" in orm_tables:
+                query_tech = query_tech.join(
+                    orm_tables["kwk_data"],
+                    orm.BasicUnit.KwkMastrNummer == orm_tables["kwk_data"].KwkMastrNummer,
+                    isouter=True,
+                )
+            if "permit_data" in orm_tables:
+                query_tech = query_tech.join(
+                    orm_tables["permit_data"],
+                    orm.BasicUnit.GenMastrNummer == orm_tables["permit_data"].GenMastrNummer,
+                    isouter=True,
+                )
+
+            # Restricted to technology
+            query_tech = query_tech.filter(
+                orm.BasicUnit.Einheittyp == unit_type_map_reversed[tech]
+            )
+
+            # Limit returned rows of query
+            if limit:
+                query_tech = query_tech.limit(limit)
+
+            return query_tech
+
+        if additional_table:
+
+            orm_table = getattr(orm, ORM_MAP[additional_table], None)
+
+            query_additional_tables = Query(orm_table, session=session)
+
+            # Limit returned rows of query
+            if limit:
+                query_additional_tables = query_additional_tables.limit(limit)
+
+            return query_additional_tables
+
+def save_metadata(data:list = None, engine=None) -> None:
+    """
+    Save metadata during csv export.
+
+    Parameters
+    ----------
+    data: list
+        List of exported technologies for which metadata is needed.
+    engine: <class 'sqlalchemy.engine.base.Engine'>
+        User-defined database engine.
+
+    Returns
+    -------
+
+    """
+    data_path = get_data_version_dir()
+    filenames = get_filenames()
+    metadata_file = os.path.join(data_path, filenames["metadata"])
+    unit_type_map_reversed = reverse_unit_type_map()
+
+    with session_scope(engine=engine) as session:
+
+        # check for latest db entry for exported technologies
+        mastr_technologies = [
+            unit_type_map_reversed[tech] for tech in data
+        ]
+        newest_date = (
```

Comment:
This is rather because `technologies_to_export` is emtpy in case of additional_tables and the query fails. The backfill is performed before in https://github.com/OpenEnergyPlatform/open-MaStR/blob/46f51056636404e2c706ab9eb69bf164083f4d54/open_mastr/mastr.py#L341-L344

However, thanks for spotting. This will be tackled in https://github.com/OpenEnergyPlatform/open-MaStR/issues/386 and is already listed as action point there. Until then, I comment `save_metadata` out.

Reply:
Resolved with https://github.com/OpenEnergyPlatform/open-MaStR/pull/401/commits/dc883995de0ec41e678ddccc67f644bc5d21da68 at the moment.